sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -386,24 +386,44 @@ public Expression field(BlockBuilder list, int index, Type storageType) {
       }
       Expression field = Expressions.call(expression, getter, Expressions.constant(index));
       if (fromType.getTypeName().isLogicalType()) {
-        field = Expressions.call(field, "getMillis");
+        Expression millisField = Expressions.call(field, "getMillis");
         String logicalId = fromType.getLogicalType().getIdentifier();
         if (logicalId.equals(TimeType.IDENTIFIER)) {
-          field = Expressions.convert_(field, int.class);
+          field =
+              Expressions.condition(
```

Comment:
I'm surprised calcite doesn't have a helper function for this pattern. It might be nice to add one here.

Reply:
Are you suggesting the helper function can be added in calcite git ?

Reply:
I'm suggesting you add one to this class.

```
private static Expression nullOr(Expression field, Expression ifNotNull) {
    return Expressions.condition(
        Expressions.equal(field, Expressions.constant(null)),
        Expressions.constant(null),
        Expressions.box(ifNotNull));
}
```

Then this becomes
```
nullOr(field, Expressions.convert_(millisField, int.class));
```
and the same pattern can be applied below.